File: sandbox/tests/portscan.exs
for port <- [22, 80, 443] do
  case :gen_tcp.connect(~c"127.0.0.1", port, [], 1000) do
    {:ok, sock} ->
      IO.puts("#{port} open")
      :gen_tcp.close(sock)
    _ -> nil
  end
end

File: sandbox/tests/portscan.ml
let scan port =
  try
    let sock = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
    Unix.connect sock (Unix.ADDR_INET (Unix.inet_addr_of_string "127.0.0.1", port));
    Printf.printf "%d open\n" port;
    Unix.close sock
  with _ -> ()

let () = List.iter scan [22; 80; 443]
